mock_tests/test_collection.py

```diff
@@ -339,3 +341,45 @@ def test_integration_config(
 
     client.collections.list_all()  # return is irrelevant
     weaviate_no_auth_mock.check_assertions()
+
+
+def test_year_zero(weaviate_no_auth_mock: HTTPServer, start_grpc_server: grpc.Server) -> None:
+    zero_date: properties_pb2.Value.date_value = properties_pb2.Value(
+        date_value="0000-01-30T00:00:00Z"
+    )
+    date_prop: Mapping[str, properties_pb2.Value.date_value] = {"date": zero_date}
+
+    class MockWeaviateService(weaviate_pb2_grpc.WeaviateServicer):
+        def Search(
+            self, request: search_get_pb2.SearchRequest, context: grpc.ServicerContext
+        ) -> search_get_pb2.SearchReply:
+            return search_get_pb2.SearchReply(
+                results=[
+                    search_get_pb2.SearchResult(
+                        properties=search_get_pb2.PropertiesResult(
+                            non_ref_props=properties_pb2.Properties(fields=date_prop)
+                        )
+                    ),
+                ]
+            )
+
+    weaviate_pb2_grpc.add_WeaviateServicer_to_server(MockWeaviateService(), start_grpc_server)
```

Comment:
For adding new mocking abilities, would it be better placed in `conftest` alongside `start_grpc_server`? I wonder whether adding services to the server inside tests might accidentally clobber other tests that depend on services added in `conftest`, if that makes sense

This isn't a concern now but perhaps might be as we grow the mocking capabilities. How would it be best to mock a gRPC server where we have a single gRPC method that serves many different use-cases without just basically reinveting the actual server inside the mock? 😭

Maybe down the line we can have a `client_factory` that handles its own mock gRPC server within it and adds the services necessary there so that the mocked server doesn't get altered between specific tests

Reply:
> For adding new mocking abilities, would it be better placed in conftest alongside start_grpc_server? I wonder whether adding services to the server inside tests might accidentally clobber other tests that depend on services added in conftest, if that makes sense

hmmmm, not sure - I would wait until we have more of these tests to see if we can "generalize" them. Is that ok with you?

Reply:
100%, the above was just a comment about what to do moving forward and didn't need fixing in this PR :D